crowbar_framework/app/models/service_object.rb

```diff
@@ -1125,6 +1125,12 @@ def apply_role(role, inst, in_queue)
 
       save_it = false
 
+      # reset possible previous error flags at the node
```

Comment:
So I'd remove this for now since I guess it needs some more thinking. Since you reset the attribute in CrowbarService already, we should be safe for now.